lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -188,8 +195,46 @@ public DataLoadingRequestBuilder setInputFormatClass(
     return this;
   }
 
+  /**
+   * Sets the path of the folder where the data is.
+   * Set to ANY the evaluator that will be able to load this data.
+   *
+   * @deprecated since 0.12. Should use instead
+   *             {@link DataLoadingRequestBuilder#addDataPartition(DataPartition)}
+   *             or {@link DataLoadingRequestBuilder#addDataPartitions(List)}
+   * @param inputPath
+   *          the input path
+   * @return this
+   */
+  @Deprecated
   public DataLoadingRequestBuilder setInputPath(final String inputPath) {
-    this.inputPath = inputPath;
+    this.partitions = new ArrayList<>(Arrays.asList(new DataPartition(inputPath, DataPartition.ANY)));
```

Comment:
Is this safe? Shouldn't you check that the list is empty at this time? In that case you just call `this.partitions.add()`. 

Also, we need to check that someone using this is *only* using this, not the `addDataPartitions` method as well. Otherwise, we might override what they are doing.